src/CrossSigningManager.js

```diff
@@ -132,24 +132,14 @@ const onSecretRequested = async function({
     if (name.startsWith("m.cross_signing")) {
         const callbacks = client.getCrossSigningCacheCallbacks();
         if (!callbacks.getCrossSigningKeyCache) return;
-        /* Explicit enumeration here is deliberate – never share the master key! */
-        if (name === "m.cross_signing.self_signing") {
-            const key = await callbacks.getCrossSigningKeyCache("self_signing");
-            if (!key) {
-                console.log(
-                    `self_signing requested by ${deviceId}, but not found in cache`,
-                );
-            }
-            return key && encodeBase64(key);
-        } else if (name === "m.cross_signing.user_signing") {
-            const key = await callbacks.getCrossSigningKeyCache("user_signing");
-            if (!key) {
-                console.log(
-                    `user_signing requested by ${deviceId}, but not found in cache`,
-                );
-            }
-            return key && encodeBase64(key);
+        const keyId = name.replace("m.cross_signing.", "");
```

Comment:
Wondering if an explicit list would still be a good idea here, just in case we add something to the cache in the future? Perhaps that's not very likely though...

Reply:
Ah yeah, may as well, I'll add one back in.